syntax = "proto3";
package input_output;


// HTTP_Response is the basic response for every HTTP request a user or API performs
message HTTP_Response
{
    // success is boolean response that is returned stating if the request was successful or had failed
    bool success = 1;
    // http_status displays the relevant code regarding the success of the specified request made
    int32 http_status = 2;
    // message is a short summary of the response
    string message = 3;
    // error contains all the error messages that may occur when performing the specified request
    repeated string error = 4;
}

// UpdateUserEmailRequest
message OTP_Response {
    // status
    HTTP_Response status = 1;
    // otp_required
    bool otp_required = 2;
}
